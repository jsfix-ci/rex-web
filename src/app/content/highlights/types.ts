import { Highlight, HighlightColorEnum } from '@openstax/highlighter/dist/api';

export type HighlightData = Highlight;
export interface SummaryHighlights {
  [sectionId: string]: {[pageId: string]: HighlightData[]};
}
export interface SummaryFilters {
  chapters: string[];
  colors: HighlightColorEnum[];
}

export interface State {
  myHighlightsOpen: boolean;
  enabled: boolean;
  focused?: string;
  highlights: null | HighlightData[];
  summary: {
    filters: SummaryFilters,
    loading: boolean;
    chapterCounts: {[key: string]: number};
    highlights: SummaryHighlights;
  };
}
