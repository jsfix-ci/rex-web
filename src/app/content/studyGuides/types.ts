import { Highlight } from '@openstax/highlighter/dist/api';
import { HighlightsSummary } from '@openstax/highlighter/dist/api';
import { LinkedArchiveTreeNode } from '../types';

export type CountsPerSource = NonNullable<HighlightsSummary['countsPerSource']>;
export type StudyGuidesPagination = null | {
  sourceIds: string[];
  page: number;
  perPage: number;
};

export type OrderedSummaryHighlights = Array<{
  location: LinkedArchiveTreeNode,
  pages: Array<{
    pageId: string;
    highlights: Highlight[];
  }>
}>;

export interface StudyGuides {
  [locationId: string]: {[pageId: string]: Highlight[]};
}

export interface State {
  highlights: StudyGuides | null;
  isEnabled: boolean;
  loading: boolean;
  open: boolean;
  pagination: StudyGuidesPagination;
  summary: HighlightsSummary | null;
  totalCountsPerPage: CountsPerSource | null;
}
