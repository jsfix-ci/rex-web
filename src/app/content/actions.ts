import {createStandardAction} from 'typesafe-actions';

export const openToc = createStandardAction('Content/openToc')<void>();
export const closeToc = createStandardAction('Content/closeToc')<void>();
