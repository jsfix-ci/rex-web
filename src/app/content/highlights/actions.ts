import { NewHighlight, UpdateHighlightRequest, HighlightColorEnum } from '@openstax/highlighter/dist/api';
import { createStandardAction } from 'typesafe-actions';
import { HighlightData, SummaryHighlights } from './types';

export const focusHighlight = createStandardAction('Content/Highlight/focus')<string>();
export const clearFocusedHighlight = createStandardAction('Content/Highlight/clear')();
export const createHighlight = createStandardAction('Content/Highlight/create')<NewHighlight & {id: string}>();
export const deleteHighlight = createStandardAction('Content/Highlight/delete')<string>();
export const updateHighlight = createStandardAction('Content/Highlight/update')<UpdateHighlightRequest>();
export const receiveHighlights = createStandardAction('Content/Highlight/receive')<HighlightData[]>();

export const openMyHighlights = createStandardAction('Content/openMyHighlights')<void>();
export const closeMyHighlights = createStandardAction('Content/closeMyHighlights')<void>();

export const setColorsFilter = createStandardAction('Content/setColorsFilter')<HighlightColorEnum[]>();
export const setChaptersFilter = createStandardAction('Content/setChaptersFilter')<string[]>();
export const filtersChange = createStandardAction('Content/filtersChange')<void>();
export const setIsLoadingSummary = createStandardAction('Content/setIsLoadingSummary')<boolean>();
export const receiveSummaryHighlights = createStandardAction('Content/receiveSummaryHighlights')<SummaryHighlights>();
