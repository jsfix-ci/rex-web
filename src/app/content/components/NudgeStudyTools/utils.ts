import { HTMLElement } from '@openstax/types/lib.dom';
import * as Cookies from 'js-cookie';
import React from 'react';
import { useSelector } from 'react-redux';
import { useDebouncedWindowSize } from '../../../reactUtils';
import { assertDocument, remsToPx } from '../../../utils';
import { page as pageSelector } from '../../selectors';
import { hasStudyGuides } from '../../studyGuides/selectors';
import { toolbarButtonMargin } from '../constants';
import {
  arrowDesktopHeight,
  arrowLeftMargin,
  arrowMobileHeight,
  arrowTopMargin,
  closeButtonDistanceFromContent,
  contentMarginTop,
  cookieNudgeStudyGuidesCounter,
  cookieNudgeStudyGuidesDate,
  cookieNudgeStudyGuidesPageCounter,
  nudgeStudyToolsMinPageLimit,
  nudgeStudyToolsShowLimit,
  nudgeStudyToolsTargetId,
  spotlightPadding,
  timeIntervalBetweenShowingNudgeInMs,
} from './constants';

interface Positions {
  arrowLeft: number;
  arrowTopOffset: number;
  closeButtonLeft: number;
  closeButtonTopOffset: number;
  contentWrapperRight: number;
  contentWrapperTopOffset: number;
  spotlightHeight: number;
  spotlightLeftOffset: number;
  spotlightTopOffset: number;
  spotlightWidth: number;
}

export const usePositions = (isMobile: boolean) => {
  const [windowWidth] = useDebouncedWindowSize();
  const [positions, setPositions] = React.useState<Positions | null>(null);
  const target = useGetStudyToolsTarget();

  React.useEffect(() => {
    if (target) {
      const { top, left, right, height, width } = target.getBoundingClientRect();
      const padding = remsToPx(spotlightPadding);
      const spotlightTopOffset = top - padding;
      const spotlightLeftOffset = left - padding;
      const spotlightHeight = height + (padding * 2);
      const spotlightWidth = width + (padding * 2) - (isMobile ? 0 : remsToPx(toolbarButtonMargin));
      const arrowTopOffset = spotlightTopOffset + spotlightHeight + remsToPx(arrowTopMargin);
      const arrowLeft = spotlightLeftOffset + remsToPx(arrowLeftMargin);
      const contentWrapperTopOffset = arrowTopOffset
        + remsToPx(isMobile ? arrowMobileHeight : arrowDesktopHeight)
        + remsToPx(contentMarginTop);
      const contentWrapperRight = windowWidth - right - padding;
      const closeButtonLeft = left + spotlightWidth;
      const closeButtonTopOffset = contentWrapperTopOffset - remsToPx(closeButtonDistanceFromContent);

      setPositions({
        arrowLeft,
        arrowTopOffset,
        closeButtonLeft,
        closeButtonTopOffset,
        contentWrapperRight,
        contentWrapperTopOffset,
        spotlightHeight,
        spotlightLeftOffset,
        spotlightTopOffset,
        spotlightWidth,
      });
    }
    return () => setPositions(null);
  }, [target, windowWidth, isMobile]);

  return positions;
};

const useGetStudyToolsTarget = () => {
  const document = assertDocument();
  const [target, setTarget] = React.useState<HTMLElement | null>(null);
  const studyGuides = useSelector(hasStudyGuides);

  React.useEffect(() => {
    if (studyGuides) {
      setTarget(document.querySelector(`#${nudgeStudyToolsTargetId}`) as HTMLElement | null);
    }
    return () => setTarget(null);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [studyGuides]);

  return target;
};

export const getCounterCookie = () => {
  return Number(Cookies.get(cookieNudgeStudyGuidesCounter) || 0);
};

export const getDateCookie = () => {
  const lastShownDate = Cookies.get(cookieNudgeStudyGuidesDate);
  return lastShownDate ? new Date(lastShownDate) : undefined;
};

export const passedTimeInterval = () => {
  const now = new Date();
  const lastShownDate = getDateCookie();
  return !lastShownDate
    ? true
    : (now.getTime() - lastShownDate.getTime()) > timeIntervalBetweenShowingNudgeInMs
  ;
};

export const getPageCounterCookie = () => {
  return Number(Cookies.get(cookieNudgeStudyGuidesPageCounter) || 0);
};

export const incrementPageCounterCookie = () => {
  const counter = getPageCounterCookie();
  Cookies.set(cookieNudgeStudyGuidesPageCounter, (counter + 1).toString());
};

export const shouldDisplayNudgeStudyTools = (): boolean => {
  const counter = getCounterCookie();
  const numberOfPagesOpenedByUser = getPageCounterCookie();

  return counter < nudgeStudyToolsShowLimit
    && passedTimeInterval()
    && (numberOfPagesOpenedByUser >= nudgeStudyToolsMinPageLimit);
};

export const setNudgeStudyToolsCookies = () => {
  const now = new Date();
  const counter = getCounterCookie();
  Cookies.set(cookieNudgeStudyGuidesCounter, (counter + 1).toString());
  Cookies.set(cookieNudgeStudyGuidesDate, now.toString());
  Cookies.remove(cookieNudgeStudyGuidesPageCounter);
};

export const useIncrementPageCounter = () => {
  const page = useSelector(pageSelector);
  return React.useEffect(() => {
    const counter = getPageCounterCookie();
    if (page && counter <= nudgeStudyToolsMinPageLimit) { incrementPageCounterCookie(); }
  }, [page]);
};
