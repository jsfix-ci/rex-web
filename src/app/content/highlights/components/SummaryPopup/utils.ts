import { Highlight } from '@openstax/highlighter/dist/api';
import { generateTargetQueryFromScrollTarget } from '../../../../navigation/utils';
import { Book } from '../../../types';
import { findArchiveTreeNode } from '../../../utils/archiveTreeUtils';
import { getBookPageUrlAndParams } from '../../../utils/urlUtils';

export const createHighlightLink = ({ sourceId, id, anchor: elementId }: Highlight, book?: Book) => {
  const page = book ? findArchiveTreeNode(book.tree, sourceId) : undefined;
  const target = generateTargetQueryFromScrollTarget({ id, type: 'highlight', elementId });
  return `${page && book ? getBookPageUrlAndParams(book, page).url : ''}?${target}`;
};
