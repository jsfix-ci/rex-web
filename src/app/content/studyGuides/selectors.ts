import { createSelector } from 'reselect';
import { HighlightLocationFilters } from '../highlights/types';
import { getHighlightLocationFilters, getSortedSummaryHighlights } from '../highlights/utils';
import * as parentSelectors from '../selectors';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.studyGuides
);

export const studyGuidesEnabled = createSelector(
  localState,
  (state) => state.isEnabled
);

export const studyGuidesSummary = createSelector(
  localState,
  (state) => state.summary
);

export const studyGuidesSummaryIsNotEmpty = createSelector(
  studyGuidesSummary,
  (summary) => summary !== null
    && summary.countsPerSource
    && Object.keys(summary.countsPerSource).length > 0
);

export const studyGuidesOpen = createSelector(
  localState,
  (state) => state.open
);

export const studyGuidesIsLoading = createSelector(
  localState,
  (state) => state.loading
);

export const studyGuidesPagination = createSelector(
  localState,
  (state) => state.pagination
);

export const totalCountsPerPage = createSelector(
  localState,
  (state) => state.totalCountsPerPage
);

export const hasMoreResults = createSelector(
  studyGuidesPagination,
  (pagination) => Boolean(pagination)
);

export const studyGuidesHighlights = createSelector(
  localState,
  (state) => state.highlights
);

export const highlightLocationFilters = createSelector(
  parentSelectors.book,
  (book) => book
    ? getHighlightLocationFilters(book)
    : new Map() as HighlightLocationFilters
);

export const orderedStudyGuidesHighlights = createSelector(
  studyGuidesHighlights,
  highlightLocationFilters,
  (highlightsToSort, locationFilters) => {
    return getSortedSummaryHighlights(highlightsToSort, locationFilters);
  }
);
