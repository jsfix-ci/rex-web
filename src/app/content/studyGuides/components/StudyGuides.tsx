import { HTMLElement } from '@openstax/types/lib.dom';
import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';
import { typesetMath } from '../../../../helpers/mathjax';
import Loader from '../../../components/Loader';
import { useServices } from '../../../context/Services';
import theme from '../../../theme';
import { assertWindow } from '../../../utils';
import allImagesLoaded from '../../components/utils/allImagesLoaded';
// Temporary import from /highlights directory until we make all this logic reusable and move it to content/
import * as Styled from '../../highlights/components/ShowMyHighlightsStyles';
import { SectionHighlights } from '../../highlights/components/SummaryPopup/SectionHighlights';
import { HighlightsChapterWrapper, HighlightSection } from '../../highlights/components/SummaryPopup/styles';
import * as selectors from '../selectors';

// Why these styles are not applied?
// tslint:disable-next-line: variable-name
const StyledSectionHighlights = styled(SectionHighlights)`
  ${HighlightsChapterWrapper} {
    ${theme.breakpoints.mobile`
      max-width: 90%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `}
  }
  ${HighlightSection} {
    ${theme.breakpoints.mobile`
    padding-left: 2rem;
  `}
  }
`;

// tslint:disable-next-line: variable-name
const StudyGuides = () => {
  const orderedHighlights = useSelector(selectors.orderedStudyGuidesHighlights);
  const isLoading = useSelector(selectors.studyGuidesIsLoading);
  const container = React.useRef<HTMLElement>(null);
  const services = useServices();

  React.useLayoutEffect(() => {
    if (container.current) {
      services.promiseCollector.add(allImagesLoaded(container.current));
      services.promiseCollector.add(typesetMath(container.current, assertWindow()));
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps, ignore promiseCollector
  }, [orderedHighlights]);

  return <React.Fragment>
    {isLoading ? <Styled.LoaderWrapper><Loader large /></Styled.LoaderWrapper> : null}
    {orderedHighlights && <Styled.Highlights ref={container}>
      {orderedHighlights.map((highlightData) => {
        return <StyledSectionHighlights
          key={highlightData.location.id}
          highlightDataInSection={highlightData}
          forStudyGuides={true}
        />;
      })}
    </Styled.Highlights>}
  </React.Fragment>;
};

export default StudyGuides;
