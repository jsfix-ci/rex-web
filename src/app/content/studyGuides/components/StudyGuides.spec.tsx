import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestServices from '../../../../test/createTestServices';
import createTestStore from '../../../../test/createTestStore';
import { book as archiveBook, page, pageInChapter } from '../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../test/mocks/osWebLoader';
import * as Services from '../../../context/Services';
import MessageProvider from '../../../MessageProvider';
import { Store } from '../../../types';
import { receiveBook, receivePage } from '../../actions';
// Temporary import from /highlights directory until we make all this logic reusable and move it to content/
import { highlightLocationFilters } from '../../highlights/selectors';
import { getHighlightLocationFilterForPage } from '../../highlights/utils';
import { formatBookData } from '../../utils';
import { stripIdVersion } from '../../utils/idUtils';
import { receiveStudyGuidesHighlights } from '../actions';
import { StudyGuidesHighlights } from '../types';
import StudyGuides from './StudyGuides';

const hlBlue = { id: 'hl1', color: HighlightColorEnum.Blue, annotation: 'hl1' };
const hlGreen = { id: 'hl2', color: HighlightColorEnum.Green, annotation: 'hl' };
const hlPink = { id: 'hl3', color: HighlightColorEnum.Pink, annotation: 'hl' };
const hlPurple = { id: 'hl4', color: HighlightColorEnum.Purple, annotation: 'hl' };
const hlYellow = { id: 'hl5', color: HighlightColorEnum.Yellow };

describe('StudyGuides', () => {
  const book = formatBookData(archiveBook, mockCmsBook);
  let store: Store;
  let services: ReturnType<typeof createTestServices>;

  beforeEach(() => {
    store = createTestStore();

    store.dispatch(receiveBook(book));
    store.dispatch(receivePage({...page, references: []}));

    services = createTestServices();
  });

  it('properly display summary highlights', () => {
    const state = store.getState();
    const pageId = stripIdVersion(page.id);
    const locationFilters = highlightLocationFilters(state);
    const location = getHighlightLocationFilterForPage(locationFilters, pageInChapter);

    const summaryHighlights = {
      [pageId]: {
        [pageId]: [hlBlue, hlGreen, hlPink, hlPurple, hlYellow],
      },
      [location!.id]: {
        [pageInChapter.id]: [hlBlue, hlGreen],
      },
    } as StudyGuidesHighlights;

    store.dispatch(receiveStudyGuidesHighlights(summaryHighlights));

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <StudyGuides />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
