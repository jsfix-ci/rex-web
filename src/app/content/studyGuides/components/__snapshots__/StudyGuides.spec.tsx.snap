// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyGuides properly display summary highlights 1`] = `
<div
  className="ShowMyHighlightsStyles__Highlights-idlhpx-0 hSwPWo"
>
  <div
    className="styles__HighlightsChapterWrapper-sc-1t1yajj-1 bbDwHa"
  >
    <div
      className="styles__HighlightsChapter-sc-1t1yajj-2 jeAUlf"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"os-text\\">Test Page 1</span>",
        }
      }
      data-testid="chapter-title"
    />
  </div>
  <div
    className="styles__HighlightWrapper-sc-1t1yajj-3 irQJfY"
  >
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl1
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 iiFIjG"
        color="blue"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl1"
        />
      </div>
    </div>
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 hNuDha"
        color="green"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl2"
        />
      </div>
    </div>
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 kkHNUJ"
        color="pink"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl3"
        />
      </div>
    </div>
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 zYAIs"
        color="purple"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl4"
        />
      </div>
    </div>
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      />
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 gjNPey"
        color="yellow"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl5"
        />
      </div>
    </div>
  </div>
  <div
    className="styles__HighlightsChapterWrapper-sc-1t1yajj-1 bbDwHa"
  >
    <div
      className="styles__HighlightsChapter-sc-1t1yajj-2 jeAUlf"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"os-number\\">10</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
        }
      }
      data-testid="chapter-title"
    />
  </div>
  <div
    className="styles__HighlightWrapper-sc-1t1yajj-3 irQJfY"
  >
    <div
      className="styles__HighlightSection-sc-1t1yajj-4 iBRaQY"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"os-text\\">Test Page 6</span>",
        }
      }
      data-testid="section-title"
    />
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl1
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 iiFIjG"
        color="blue"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl1"
        />
      </div>
    </div>
    <div
      className="StudyGuidesListElement__HighlightOuterWrapper-znhmjn-0 bydcVT"
    >
      <div
        className="StudyGuidesListElement__HighlightAnnotation-znhmjn-1 cEpfVc"
      >
        hl
      </div>
      <div
        className="StudyGuidesListElement__HighlightContentWrapper-znhmjn-3 hNuDha"
        color="green"
      >
        <div
          className="summary-highlight-content StudyGuidesListElement__HighlightContent-znhmjn-2 jqbHrH"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-highlight-id="hl2"
        />
      </div>
    </div>
  </div>
</div>
`;
